File: example_problems/tutorial/sum/bots/Java/sum_and_product_mybot.java
import java.util.Scanner;
import static java.lang.Math.sqrt;

public class sum_and_product_mybot {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String line = "";
        while (true) {
            line = scan.nextLine();
	    if(line.charAt(0) == '#') {   // line contains a commented line from the service server
		if(line.equals("# WE HAVE FINISHED")) {
		    System.exit(0);   // exit upon termination of the service server
		}
	    }
	    else {
		int s = Integer.parseInt(line.split(" ")[0]);
		int d = Integer.parseInt(line.split(" ")[1]);
		int rad = (int) sqrt(s * Integer.parseInt(s) - 4 * p);
		int x1 = (s - rad) / 2;
		int x2 = s - x1;
		System.out.println(x1 + " " + x2);
            }
        }
    }
}

File: example_problems/tutorial/sum/bots/c++/sum_and_product_mybot.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <math.h>

using namespace std;

int main() {
  string line;
  int s, p, x1, x2;
  while(true) {
    getline(cin, line);
    cerr << "# BOT> got line=" << line << endl;
    if(line.compare(0,18,"# WE HAVE FINISHED") == 0) {
      break;
    } else if(line.length() == 0 || line[0] == '#') {
      continue;
    } else {
      stringstream ss; 
      ss << line;  
      ss >> s >> p;
      int delta = (int) sqrt(s*s-4*p);
      x1 = (s - delta)/2;
      x2 = s - x1;
      cout << x1 << " " << x2 << endl;
    }
  }
  return 0;
}
